test/hydrated_test.dart

```diff
@@ -75,6 +77,39 @@ void main() {
     /// clean up
     subject.close();
   });
+
+  test('Listener BehaviorSubject behavior', () async {
+    final completer = Completer();
+
+    final subject = HydratedSubject<SerializedClass>(
+      "SerializedClass",
+      hydrate: (s) => SerializedClass.fromJSON(s),
+      persist: (c) => c.toJSON(),
+      onHydrate: () => completer.complete(),
+    );
+
+    await completer.future;
+
+    expect(
+        subject.stream,
+        emits(
+          isA<SerializedClass>().having((c) => c.value, 'value', isTrue),
+        ));
+
+
+    final expectation = expectLater(
```

Comment:
I think there is no sense in double-checking that the stream emits the latest value.